azure-proto-core/Resources/Plan.cs

```diff
@@ -112,34 +47,16 @@ public bool Equals(Plan other)
                 return false;
             }
 
-            if (Name == null && other.Name != null || Name != null && !Name.Equals(other.Name))
-            {
-                return false;
-            }
-
-            if (Product == null && other.Product != null || Product != null && !Product.Equals(other.Product))
-            {
-                return false;
-            }
-
-            if (PromotionCode == null && other.PromotionCode != null ||
-                PromotionCode != null && !PromotionCode.Equals(other.PromotionCode))
-            {
-                return false;
-            }
-
-            if (Publisher == null && other.Publisher != null ||
-                Publisher != null && !Publisher.Equals(other.Publisher))
-            {
-                return false;
-            }
-
-            if (Version == null && other.Version != null || Version != null && !Version.Equals(other.Version))
+            if (object.Equals(Name, other.Name) &&
+                object.Equals(Product, other.Product) &&
```

Comment:
here you don't need to compare individual properties. Just (this, other). This is supposed to be an optimization if this and other is same object, then true. For each properties, let it compare below.

Reply:
object.Equals I believe does a ReferenceEquals only.  If there was a string.Equals(a, b) then that could be used here and avoid some of the null checking that was previously there